src/promql/src/extension_plan/range_manipulate.rs

```diff
@@ -592,14 +597,25 @@ impl RangeManipulateStream {
 
         // calculate for every aligned timestamp (`curr_ts`), assume the ts column is ordered.
         let mut range_start_index = 0usize;
+        let mut last_range_start = 0;
+        let mut start_delta = 0;
```

Comment:
non-blocking: It'd be better if we can document how they work.

Reply:
document around where they are used